napari/_qt/qt_resources/_svg.py

```diff
@@ -0,0 +1,149 @@
+    >>> label = QLabel()
+    >>> icon = QColoredSVGIcon.from_resources('new_points')
+    >>> label.setPixmap(icon.colored(theme='light').pixmap(300, 300))
+    >>> label.show()
+    """
+
+    def __init__(
+        self, path_or_xml: str, color: str = None, opacity: float = 1.0
+    ) -> None:
+        from ...resources import get_colorized_svg
+
+        self._svg = path_or_xml
+        colorized = get_colorized_svg(path_or_xml, color, opacity)
+        super().__init__(SVGBufferIconEngine(colorized))
+
+    @lru_cache()
+    def colored(
+        self,
+        color: str = None,
+        opacity: float = 1.0,
+        theme: str = None,
+        theme_key: str = 'icon',
+    ):
+        """Return a new colorized QIcon instance.
+
+        Parameters
+        ----------
+        color : str, optional
+            A valid CSS color string, used to colorize the SVG.  If provided,
+            will take precedence over ``theme``, by default None.
+        opacity : float, optional
+            Fill opacity for the icon (0-1).  By default 1 (opaque).
+        theme : str, optional
+            Name of the theme to from which to get `theme_key` color.
+            ``color`` argument takes precedence.
+        theme_key : str, optional
+            If using a theme, key in the theme dict to use, by default 'icon'
+
+        Returns
+        -------
+        QColoredSVGIcon
+            A pre-colored QColoredSVGIcon (which may still be recolored)
+        """
+        if not color and theme:
+            from ...utils.theme import get_theme
+
+            color = get_theme(theme)[theme_key]
+
+        return QColoredSVGIcon(self._svg, color, opacity)
+
+    @classmethod
+    @lru_cache()
+    def from_resources(
+        cls,
+        icon_name: str,
+    ):
+        """Get an icon from napari SVG resources.
+
+        Parameters
+        ----------
+        icon_name : str
+            The name of the icon svg to load (just the stem).  Must be in the
+            napari icons folder.
+
+        Returns
+        -------
+        QColoredSVGIcon
+            A colorizeable QIcon
+        """
+        from ...resources import get_icon_path
+
+        path = get_icon_path(icon_name)
+        return QColoredSVGIcon(path)
+
+
+class SVGBufferIconEngine(QIconEngine):
+    """A custom QIconEngine that can render an SVG buffer.
+
+    An icon engine provides the rendering functions for a ``QIcon``.
+    Each icon has a corresponding icon engine that is responsible for drawing
+    the icon with a requested size, mode and state.  While the built-in
+    QIconEngine is capable of rendering SVG files, it's not able to receive the
+    raw XML string from memory.
+
+    This ``QIconEngine`` takes in SVG data as a raw xml string or bytes.
+
+    see: https://doc.qt.io/qt-5/qiconengine.html
+    """
+
+    def __init__(self, xml: str | bytes) -> None:
+        if isinstance(xml, str):
+            xml = bytes(xml, encoding='utf-8')
```

Comment:
Oh, I've never seen `bytes(...)` used like that before; any difference with `.encode()`  beyond performance ? 
I doubt it matters though.
```
In [20]: %timeit '123'.encode('utf-8')
75.1 ns ± 3.26 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [21]: %timeit bytes('123', 'utf-8')
186 ns ± 40.6 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)
```

Reply:
TIL.  will change